src/components/NewMessageForm/NewMessageForm.vue

```diff
@@ -285,7 +285,9 @@ export default {
 			for (const index in shareableFiles) {
 				// The pat of the file to share to the conversation
 				const path = this.$store.getters.getAttachmentFolder() + '/' + shareableFiles[index].name
-				shareFile(path, token)
+				await shareFile(path, token)
+				// Mark current file as shared
+				this.$store.dispatch('markFileasShared', { token, index })
```

Comment:
Concurrency problem. Need to update it to sharing before doing the await similar as with uploading